src/roiextractors/extractors/memmapextractors/numpymemampextractor.py

```diff
@@ -0,0 +1,178 @@
+
+    extractor_name = "NumpyMemmapImagingExtractor"
+
+    def __init__(
+        self,
+        file_path: PathType,
+        frame_shape: Tuple[int, int],
+        sampling_frequency: float,
+        dtype: DtypeType,
+        offset: int = 0,
+        image_structure_to_axis: Dict[str, int] = None,
+    ):
+        """Class for reading optical imaging data stored in a binary format with np.memmap
+
+
+        Parameters
+        ----------
+        file_path : PathType
+            the file_path where the data resides.
+        frame_shape : tuple
+            The frame shape of the image determines how each frame looks. Examples:
+            (n_channels, rows, columns), (rows, columns, n_channels), (n_channels, columns, rows), etc.
+            Note that n_channels is 1 for grayscale and 3 for color images.
+        dtype : DtypeType
+            The type of the data to be loaded (int, float, etc.)
+        offset : int, optional
+            The offset in bytes. Usually corresponds to the number of bytes occupied by the header. 0 by default.
+        sampling_frequency : float, optional
+            The sampling frequency.
+        image_structure_to_axis : dict, optional
+            A dictionary indicating what axis corresponds to what in the memmap. The default values are:
+            dict(frame_axis=0, num_channels=1, rows=2, columns=3)
+            frame_axis=0 indicates that the first axis corresponds to the frames (usually time)
+            num_channels=1 here indicates that the first axis corresponds to the  n_channels.
+            rows=2 indicates that the rows in the image are in the second axis
+            columns=3 indicates that columns is the last axis or dimension in this structure.
+
+            Notice that this should correspond with frame_shape.
+        """
+
+        self.installed = True
+        super().__init__()
+
+        self.file_path = Path(file_path)
+        self._sampling_frequency = sampling_frequency
+        self.offset = offset
+        self.dtype = dtype
+
+        # Get the structure, apply default if not available
+        self.frame_shape = frame_shape
+        image_structure_to_axis = dict() if image_structure_to_axis is None else image_structure_to_axis
+        self.image_structure_to_axis = dict(frame_axis=0, num_channels=1, rows=2, columns=3)
+        self.image_structure_to_axis.update(image_structure_to_axis)
+        self.frame_axis = self.image_structure_to_axis["frame_axis"]
+
+        # Extract video
+        self._video = self.read_binary_video()
+
+        # Get the image structure as attributes
+        self._rows = self._video.shape[self.image_structure_to_axis["rows"]]
+        self._columns = self._video.shape[self.image_structure_to_axis["columns"]]
+        self._num_channels = self._video.shape[self.image_structure_to_axis["num_channels"]]
+        self._num_frames = self._video.shape[self.image_structure_to_axis["frame_axis"]]
+
+    def read_binary_video(self):
```

Comment:
Wasn't this going to be a generic `extraction_tool` so that people could use it outside of the extractor API?

Reply:
I don't remember this. What we would get of supporting this? my feeling is that I rather support this well tested inside of our class as this pertains to `roieextractors` functionality. Supporting stand-alone use of the function seems to be outside of the scope of the repo. 

`spikeitnterface` does it this way but I don't know why. I am not very convince that we should follow them by default. Maybe if a second use of this function inside the library appears we can move it out of here?

Reply:
Original discussion: https://github.com/catalystneuro/roiextractors/pull/112#discussion_r877590302

>spikeinterface does it this way but I don't know why. I am not very convince that we should follow them by default. Maybe if a second use of this function inside the library appears we can move it out of here?

It's more for working directly with the file contents hands-on, which is a common use case. I personally used this helper function when we were initially exploring these files, so it can be quite useful generally.

As is, it's not a static method so can't be called unless `__init__` succeeded, but maybe you don't entirely know all of the arguments to the currently designed `__init__`, so having a helper to explore the file structure can help.

Reply:
OK, let's move this outside. Check #115 for a stand-alone implementation of the functionality.